### PR TITLE
upgrade to golang 1.18.6

### PR DESCRIPTION
**Pull request by M00nF1sh**
### Description
Upgrade to use latest Golang 1.18.6

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/aws-load-balancer-controller
 
-go 1.18
+go 1.19
```
Comment (kishorj) [inline go.mod:3]: lets stick with 1.18 for the patch release. We can upgrade to the new version at a slower pace.

Comment (M00nF1sh) [reply go.mod]: make sense

**Review by kishorj — APPROVED** [review]

**Pull request merged**